src/main/java/pl/javastart/task/TournamentStats.java

```diff
@@ -1,12 +1,72 @@
 package pl.javastart.task;
 
-import java.util.Scanner;
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.*;
 
 public class TournamentStats {
+    public static final String FIRST_NAME = "1";
+    public static final String LAST_NAME = "2";
+    public static final String SCORE = "3";
+    List<Player> players = new ArrayList<>();
 
-    void run(Scanner scanner) {
+    void run(Scanner scanner) throws IOException {
         // tutaj dodaj swoje rozwiązanie
         // użyj przekazanego scannera do wczytywania wartości
+        getPlayers(scanner);
+        chooseSortMethod(scanner);
+        save();
+        System.out.println("Dane posortowano i zapisano do pliku stats.csv");
+    }
+
+    private void chooseSortMethod(Scanner scanner) {
+        System.out.printf("Po jakim parametrze posortować? (%s - imię, %s - nazwisko, %s - wynik) \n", FIRST_NAME,
+                LAST_NAME, SCORE);
+        String choice = scanner.nextLine();
+        Comparator<Player> comparator = null;
+        boolean stopAsking = true;
+        do {
+            switch (choice) {
+                case FIRST_NAME -> comparator = new PlayerFirstNameComparator();
+                case LAST_NAME -> comparator = new PlayerLastNameComparator();
+                case SCORE -> comparator = new PlayerScoreComparator();
+                default -> {
+                    System.out.println("Wprowadzono błędną opcję. Jeszcze raz.");
+                    stopAsking = false;
+                }
+            }
+        } while (!stopAsking);
+        Collections.sort(players, comparator);
+
+        System.out.println("Sortować rosnąco czy malejąco? (1 - rosnąco, 2 - malejąco)");
+        int sortChoice = scanner.nextInt();
+        if (sortChoice == 2) {
+            Collections.reverse(players);
+        }
+    }
```

Comment:
Nie do końca o to mi chodziło.
Wydziel metodę, która tworzy komparator, ale nie sortuje kolekcję.
Metoda getPlayers() powinna zwrócić listę.
Metoda save() powinna przyjąć listę.
Metoda chooseSortMethod() powinna zwrócić komparator.